hook.h

```diff
@@ -27,7 +27,9 @@
  */
 __attribute__((__always_inline__))
 static inline enum xdp_action xdpcap_exit(struct xdp_md *ctx, struct bpf_map_def *hook, enum xdp_action action) {
```

Comment:
In that case you should make ctx and map def `void *` as well. Probably make action `int`. That's why I think trying to provide a magic header is futile.

Reply:
Good point, made the map a `void *`. `struct xdp_md` and `enum xdp_action` are defined in the kernel's `linux/bpf.h` so they should exist I think.